hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/impl/CommitUtilsWithMR.java

```diff
@@ -49,10 +50,11 @@ private CommitUtilsWithMR() {
   /**
    * Get the location of magic job attempts.
    * @param out the base output directory.
+   * @param jobUUID unique Job ID.
    * @return the location of magic job attempts.
    */
-  public static Path getMagicJobAttemptsPath(Path out) {
-    return new Path(out, MAGIC);
+  public static Path getMagicJobAttemptsPath(Path out, String jobUUID) {
+    return new Path(out, MAGIC_PATH_PREFIX + jobUUID);
```

Comment:
let's add a Precondition to check that the uuid isn't null/empty. I know it's unlikely, but if something goes very wrong then duplicate paths get in
```java
Preconditions.checkArgument(jobUUID != null && !(jobUUID.isEmpty()),
 "Invalid job ID: %s", jobUUID);
```